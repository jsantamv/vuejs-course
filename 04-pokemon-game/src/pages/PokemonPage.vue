<template>
  <h1>Quien es ese Pokemon?</h1>

  <PokemonPicture :pokemonId="4" :showpokemon="true"/>
  <PokemonOption/>

</template>

<script>
import PokemonOption from '@/components/PokemonOption';
import PokemonPicture from '@/components/PokemonPicture';
import getPokemonOptions from '@/helpers/getPokemonOptions'


console.log(getPokemonOptions());


export default {
    components: { PokemonOption, PokemonPicture }
}
</script>

<style>

</style>