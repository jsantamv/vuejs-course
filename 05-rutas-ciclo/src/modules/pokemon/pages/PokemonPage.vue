<template>
    <h1>Pokemon Page <span>#{{ id }}</span></h1>
</template>

<script>
export default {
    props: {
        id: {
            type: Number,
            requierd: true
        },
    },   
}
</script>