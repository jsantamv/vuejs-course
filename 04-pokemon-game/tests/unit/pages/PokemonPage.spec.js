import { shallowMount, mount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from '../mocks/pokemons.mock'
import { shallowReactive } from 'vue'



describe('PokemonPage', () => {
    let wrapper
    let mixPokemonArraySpy

    beforeEach(() => {
        //creamos un spy, hay que llamarlo antes de que se monte
        //cuando lo utilizamos aca debemos de limipar el spy
        //mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')

        wrapper = shallowMount(PokemonPage)
    })

    test('debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de llamar a mixPokemonArray al montar', () => {
        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        const wrapper = shallowMount(PokemonPage)

        expect(mixPokemonArraySpy).toHaveBeenCalled()

    })

    test('debe de hacer match con el snapshot cuando carga los pokemon', () => {
        //con esto monto el componente.
        const wrapper = shallowMount(PokemonPage, {
            //const wrapper = mount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[1],
                    showpokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de mostra los componentes de PokemonPincture y pokemon Options', () => {

        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[1],
                    showpokemon: true,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        //=========< pokemon picture >=========
        //PokemonPicture debe de existir
        const pokemonPicture = wrapper.find('pokemon-picture-stub')
        expect(pokemonPicture.exists()).toBeTruthy()

        //PokemonPicture attribute pokemonid === '5'        
        expect(pokemonPicture.attributes('pokemonid')).toBe('2')

        //=========< pokemon options >=========
        //PokemonOptions debe de existir        
        const pokemonOption = wrapper.find('pokemon-option-stub')
        expect(pokemonOption.exists()).toBeTruthy()

        //PokemonOptions attribute pokemons tobe true
        expect(pokemonOption.attributes('pokemons')).toBeTruthy()


    })
})