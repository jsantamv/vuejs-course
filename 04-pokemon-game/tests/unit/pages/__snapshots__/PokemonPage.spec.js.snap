// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage debe de hacer match con el snapshot 1`] = `
<h1>Espere por favor ...</h1>
<!--v-if-->
`;

exports[`PokemonPage debe de hacer match con el snapshot cuando carga los pokemon 1`] = `
<div>
  <h1>Quien es ese Pokemon?</h1>
  <pokemon-picture-stub pokemonid="2" showpokemon="false"></pokemon-picture-stub>
  <pokemon-option-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-option-stub>
</div>
<!--v-if-->
`;
